go/private/tools/path.bzl

```diff
@@ -81,9 +81,12 @@ def _go_path_impl(ctx):
             dst = pkg.dir + "/" + f.basename
             _add_manifest_entry(manifest_entries, manifest_entry_map, inputs, f, dst)
         for f in pkg.embedsrcs:
+            embedpath = f.path
             if src_dir == None:
                 fail("cannot relativize {}: src_dir is unset".format(f.path))
-            dst = pkg.dir + "/" + paths.relativize(f.path, src_dir)
+            if f.dirname.startswith(ctx.bin_dir.path):
```

Comment:
It's subtle, but `ctx.bin_dir` is the bin dir of the rule, which may not be the same as the bin dir of the files in `embedsrcs` when there are transitions involved. You should be able use the path of [File.root](https://bazel.build/rules/lib/File#root) instead.

Reply:
Thanks for the tip! Changed to use f.root instead.